homeassistant/components/lektrico/config_flow.py

```diff
@@ -0,0 +1,261 @@
+        self,
+        user_input: dict[str, Any] | None = None,
+        errors: dict[str, str] | None = None,
+    ) -> FlowResult:
+        """Show the setup form to the user."""
+        if user_input is None:
+            user_input = {}
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_FRIENDLY_NAME,
+                        default=user_input.get(CONF_FRIENDLY_NAME, ""),
+                    ): str,
+                    vol.Required(CONF_HOST, default=user_input.get(CONF_HOST, "")): str,
+                }
+            ),
+            errors=errors or {},
+        )
+
+    @callback
+    def _async_create_entry(self) -> FlowResult:
+        return self.async_create_entry(
+            title=f"{self._friendly_name}_{str(self._serial_number)}",
+            data={CONF_HOST: self._host, CONF_FRIENDLY_NAME: self._friendly_name},
+        )
+
+    async def async_step_zeroconf(
+        self, discovery_info: ZeroconfServiceInfo
+    ) -> FlowResult:
+        """Handle zeroconf discovery."""
+        self._host = discovery_info.host  # 192.168.100.11
+        _id = discovery_info.properties.get("id")  # 1p7k_500006
+
+        if _id is None:
+            # properties have no "id"
+            return self.async_abort(reason="missing_id")
+        _index = _id.find("_")
+        if _index == -1:
+            # "id" does not contain "_"
+            return self.async_abort(reason="missing_underline_in_id")
+        self._serial_number = _id[_index + 1 :]
+        if _id.startswith("m2w_81"):
+            self._friendly_name = lektricowifi.Device.TYPE_EM
+        elif _id.startswith("m2w_83"):
+            self._friendly_name = lektricowifi.Device.TYPE_3EM
+        else:
+            self._friendly_name = _id[:_index]  # it's the type
+
+        # Set unique id
+        await self.async_set_unique_id(self._serial_number, raise_on_progress=True)
+        # Abort if already configured, but update the last-known host
+        self._abort_if_unique_id_configured(
+            updates={CONF_HOST: self._host}, reload_on_update=True
+        )
+
+        self.context["title_placeholders"] = {
+            "serial_number": self._serial_number,
+            "friendly_name": self._friendly_name,
+        }
+
+        return await self.async_step_confirm()
+
+    async def _get_lektrico_serial_number_and_treat_unique_id(
+        self, raise_on_progress: bool = True
+    ) -> None:
+        """Get device's serial number from a Lektrico device."""
+        session = async_get_clientsession(self.hass)
+        device = lektricowifi.Device(
+            _host=self._host,
+            session=session,
+        )
+
+        settings = await device.device_config()
+
+        # Check if already configured
+        await self.async_set_unique_id(
+            settings.serial_number, raise_on_progress=raise_on_progress
+        )
```

Comment:
The unique_id must be a string. When the title is composed, the serial number is stringified; either that's needed here too, or it's not necessary to do it when making the title.